src/pfe/portal/modules/User.js

```diff
@@ -191,15 +191,15 @@ module.exports = class User {
    */
   async cancelLoad(project) {
     log.debug("cancelLoad: project " + project.projectID + " loadInProgress=" + project.loadInProgress);
-
+    this.uiSocket.emit('runloadStatusChanged', { projectID: project.projectID, status: 'cancelling' });
     if (project.loadInProgress) {
       project.loadInProgress = false;
       log.debug("Cancelling load for config: " + JSON.stringify(project.loadConfig));
-      this.uiSocket.emit('runloadStatusChanged', { projectID: project.projectID, status: 'cancelling' });
       let cancelLoadResp = await this.loadRunner.cancelRunLoad(project.loadConfig);
       this.uiSocket.emit('runloadStatusChanged', { projectID: project.projectID, status: 'cancelled' });
       return cancelLoadResp;
     }
+    this.uiSocket.emit('runloadStatusChanged', { projectID: project.projectID, status: 'cancelled' });
```

Comment:
I'm assuming sending the cancelled status even if there wasn't a run in progress to cancel is deliberate. ;-)

Reply:
yes